compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -21,6 +22,75 @@ use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use serde_with::DisplayFromStr;
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub(crate) struct CallStackId(u32);
+
+impl CallStackId {
+    pub(crate) fn root() -> Self {
+        Self::new(0)
+    }
+
+    fn new(id: usize) -> Self {
+        Self(id as u32)
+    }
+
+    pub(crate) fn index(&self) -> usize {
+        self.0 as usize
+    }
+
+    pub(crate) fn is_root(&self) -> bool {
+        self.0 == 0
+    }
+
+    // Retrieves a CallStack from a CallStackId
```

Comment:
I'm not sure "Retrieve" is right word, maybe "reconstruct"? I usually associate it with some kind of database retrieval.

Reply:
Done, I used 'construct'